client/components/PrettyPrint/PrettyPrint.jsx

```diff
@@ -0,0 +1,76 @@
+import { h, Component } from "preact";
+import cx from "classnames";
+import Prism from "prismjs";
+
+// Import prism plugins
+import "prismjs/plugins/line-numbers/prism-line-numbers";
+import "prismjs/plugins/line-highlight/prism-line-highlight";
+
+import "./prettyprint.css";
+
+const BASE_CLASS = "pretty-print";
+
+export default class PrettyPrint extends Component {
+  constructor(...args) {
+    super(...args);
+    this.highlight = this.highlight.bind(this);
```

Comment:
Yes it should work from a `class` perspective. But since this class extends from `preact`'s `Component` it does not work and `this` in this case would be the `super` class we extend from and not the `current` class(I think), so I have to explicitly bind this to the current `this` in the `constructor` for it work properly. 

PS: Had I used `class-properties` it would have auto-bound without me doing this in the `constructor`.